tracking/src/DataCollection.js

```diff
@@ -1,24 +1,31 @@
 const NOT_AVAILABLE = "Unknown"
 class DataCollection {
     constructor() {
+    }
 
+    toJson() {
+        // update data first
         this.getLanguage(); 
         this.getScreenResolution(); 
-        this.getUrls(); 
-    }
+        this.getUrls();         
 
-    toJson() {
         const res = {}; 
         res.lang = this.language; 
```

Comment:
there is no need to store into memory right?
why not just use the method `getLanguage()` and it should return a string value.

same goes for screen resolution & URLs

Reply:
changed already